ciao-launcher/main.go

```diff
@@ -237,15 +236,35 @@ func (client *agentClient) ErrorNotify(err ssntp.Error, frame *ssntp.Frame) {
 	glog.Infof("ERROR %d", err)
 }
 
-func (client *agentClient) installLauncherDeps() {
-	role := client.conn.Role()
-	osprepare.Bootstrap()
-	if role.IsNetAgent() {
-		osprepare.InstallDeps(launcherNetNodeDeps)
-	}
-	if role.IsAgent() {
-		osprepare.InstallDeps(launcherComputeNodeDeps)
+func (client *agentClient) installLauncherDeps(doneCh chan struct{}) {
+	ctx, cancelFunc := context.WithCancel(context.Background())
+
+	ch := make(chan error)
+	go func() {
+
+		osprepare.Bootstrap(ctx)
+
+		role := client.conn.Role()
+		if role.IsNetAgent() {
+			osprepare.InstallDeps(ctx, launcherNetNodeDeps)
+		}
+		if role.IsAgent() {
+			osprepare.InstallDeps(ctx, launcherComputeNodeDeps)
+		}
+
+		ch <- nil
+	}()
+
+	select {
+	case <-doneCh:
+		glog.Info("Received terminating signal.  Installation of launcher dependencies cancelled.")
```

Comment:
Right.  You still need the other case statement.  So

case <-doneCh:
  cancelFunc()
  <-ch
case err := <- ch:
  cancelFunc()
}
```